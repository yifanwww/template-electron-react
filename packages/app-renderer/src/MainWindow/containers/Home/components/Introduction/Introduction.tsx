import type { AppDetails } from '@app/common/apis/app';

import { Assets } from 'src/assets';

import css from './Introduction.module.scss';

interface IntroductionProps {
    appDetails?: AppDetails;
}

export function Introduction({ appDetails }: IntroductionProps): React.ReactNode {
    return (
        <>
            <header className={css.introduction}>
                <img className={css.logo} src={Assets.logo} alt="logo" />
                <p>
                    Edit <code>src/MainWindow/containers/Home/components/Introduction/Introduction.tsx</code> and save
                    to reload.
                </p>
                <a
                    className={css.link}
                    href="https://github.com/yifanwww/template-electron-react"
                    target="_blank"
                    rel="noopener noreferrer"
                >
                    Learn template-electron-react
                </a>
            </header>
            {appDetails && (
                <div className={css.appDetails}>
                    <code>Name: {appDetails.name}</code>
                    <code>Version: {appDetails.version}</code>
                    <code>Electron: {appDetails.module.electron}</code>
                    <code>Chrome: {appDetails.module.chrome}</code>
                    <code>Node.js: {appDetails.module.node}</code>
                    <code>V8: {appDetails.module.v8}</code>
                </div>
            )}
        </>
    );
}
