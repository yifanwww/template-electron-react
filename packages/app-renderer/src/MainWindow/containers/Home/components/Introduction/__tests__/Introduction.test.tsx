import type { AppDetails } from '@ter/app-common/apis/app';
import { render } from '@testing-library/react';

import { Introduction } from '../Introduction';

describe(`Test component \`${Introduction.name}\``, () => {
    it('should render', () => {
        const appDetails: AppDetails = {
            name: 'ter',
            version: 'unknown',
            module: {
                chrome: 'unknown',
                electron: 'unknown',
                node: 'unknown',
                v8: 'unknown',
            },
        };

        const { asFragment, getByText } = render(<Introduction appDetails={appDetails} />);
        expect(asFragment()).toMatchSnapshot();
        const linkElement = getByText(/Learn template-electron-react/i);
        expect(linkElement).toBeInTheDocument();
    });
});
