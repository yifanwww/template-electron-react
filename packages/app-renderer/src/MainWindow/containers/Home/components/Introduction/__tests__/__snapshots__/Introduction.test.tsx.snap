// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Test component \`Introduction\` should render 1`] = `
<DocumentFragment>
  <header
    class="introduction"
  >
    <img
      alt="logo"
      class="logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/MainWindow/containers/Home/components/Introduction/Introduction.tsx
      </code>
       and save to reload.
    </p>
    <a
      class="link"
      href="https://github.com/yifanwww/template-electron-react"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn template-electron-react
    </a>
  </header>
  <div
    class="appDetails"
  >
    <code>
      Name: ter
    </code>
    <code>
      Version: unknown
    </code>
    <code>
      Electron: unknown
    </code>
    <code>
      Chrome: unknown
    </code>
    <code>
      Node.js: unknown
    </code>
    <code>
      V8: unknown
    </code>
  </div>
</DocumentFragment>
`;
